sn_node/src/node/mod.rs

```diff
@@ -527,20 +510,38 @@ mod core {
             Ok(())
         }
 
-        /// Updates various state if elders changed.
-        pub(crate) async fn update_on_elder_change(
+        /// Updates various state if there was a member change.
+        pub(crate) async fn update_on_member_change(
             &mut self,
             old: &StateSnapshot,
         ) -> Result<Vec<Cmd>> {
             let new = self.state_snapshot();
 
+            let mut cmds = vec![];
+
+            {
+                // check if any adult changes
+                let current: BTreeSet<XorName> = self
+                    .network_knowledge()
+                    .adults()
+                    .iter()
+                    .map(Peer::name)
+                    .collect();
+                let added: BTreeSet<_> = current.difference(&old.adults).copied().collect();
+                let removed: BTreeSet<_> = old.adults.difference(&current).copied().collect();
+                let adults_changed = !added.is_empty() || !removed.is_empty();
+
+                if adults_changed {
```

Comment:
Is this not the same as:
```suggestion
                if current != old.adults {
```

Reply:
Yes, I believe so. Remnant from when `added`, `removed` and `remaining` where used variables 👍